src/test/groovy/org/prebid/server/functional/tests/DebugSpec.groovy

```diff
@@ -343,4 +346,51 @@ class DebugSpec extends BaseSpec {
         and: "Response should not contain ext.warnings"
         assert !response.ext?.warnings
     }
+
+    def "PBS should emit error when #reason and bidder return error"() {
+        given: "Default bid request with returnAllBidStatus and debug"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid = new Prebid(returnAllBidStatus: returnAllBidStatus, debug: debug)
+        }
+
+        and: "Default bid response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, BAD_REQUEST_400, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contain seatNonBid"
+        def seatNonBid = response.ext.prebid.seatNonBid[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == NO_BID
+
+        where:
+        returnAllBidStatus | debug | reason
+        true               | 0     | "returnAllBidStatus is true and debug 0"
+        false              | 1     | "returnAllBidStatus is false and debug 1"
+
+    }
+
+    def "PBS shouldn't emit error when return all bid status and debug flags are disable and bidder return error in response"() {
```

Comment:
Please add a test for following cases:
- bidder returns several bids for the same imp with different price. As far as I understand, seatNonBid should list losing bid.
- several bidders return bids. Losing bid should be listed as seatNonBid as far as I understand.